structures/auction.js

```diff
@@ -33,8 +33,10 @@ exports.auction = function (config = null) {
       source: s.size(s.array(s.size(s.string(), 2, 64)), 0, 1),
       target: s.size(s.array(s.size(s.string(), 2, 64)), 0, RelsMax),
       targetStatus: s.optional(s.size(s.array(s.enums(['', 'Removed', 'Disabled', 'Searching', 'NoSolution'])), 0, RelsMax)),
-      sourceName: s.optional(s.size(s.array(s.size(s.string(), 2, 128)), 0, RelsMax)),
+      sourceName: s.optional(s.size(s.array(s.size(s.string(), 2, 128)), 0, 1)),
```

Comment:
Why only 1 ?

Reply:
@fanshan : shaq for now only support one source. must be changed when multi source is developed. issue exists for 2 years but looks like it will never happened. Then better have a correct structure.